### PR TITLE
change pose to poseStamped and boolean for namedtarget

### LINKED ISSUE #5: Issues with manipulation target pose
From the robot_manager_interface we are passing a PoseStamped, but when it reaches the robot.cpp, in the moveit function it is using just the Pose.
https://github.com/temoto-framework/temoto_robot_manager/blob/rr-devel/src/robot.cpp#L558
`group_it->second->setPoseTarget(target_pose.pose);`
It should be
`group_it->second->setPoseTarget(target_pose);`

Additionally, sometimes the robot manager cannot set the right goal for the end effector (in setPoseTarget or setNamedTarget), but it still generates an empty plan making the manipulator move to its zero state, which is an undesirable behavior. 

Suggested solution: 
Validate the target pose using the boolean from the setPoseTarget and setNamedTarget 
```
if (!group_it->second->setNamedTarget(named_target))
  {
    is_plan_valid_ = false;
    throw CREATE_ERROR(temoto_core::error::Code::ROBOT_PLAN_FAIL,"Planning to named target pose '%s' failed.", named_target);
  }
```

### PR DESCRIPTION
**Pull request by FabianEP11**
Change the manipulation plan to use PoseStamped, and added an if statement to avoid execute trajectories to invalid plans
It fixes #5

### CODE REVIEW

File: src/robot.cpp [hunk 1]
```diff
@@ -555,7 +555,7 @@ void Robot::planManipulationPath(std::string& planning_group_name, const geometr
   group_it->second->setStartStateToCurrentState();
 
   // NOTE: Using Pose instead of PoseStamped because in that case it would replace the frame id of the header with empty ""
```
Comment (RValner) [inline src/robot.cpp]: I think this comment needs an update

**Review by RValner — CHANGES_REQUESTED** [review]
Just a minor thing to fix. See the comment.

File: src/robot.cpp (hunk 1 above)
Comment (FabianEP11) [reply src/robot.cpp]: I just removed it.

**Review by RValner — APPROVED** [review]
Looks good to me.

**Pull request merged**